stdlib/src/collections/dict.mojo

```diff
@@ -740,7 +740,12 @@ struct Dict[K: KeyElement, V: CollectionElement](
             var entry_value = entry[].unsafe_take()
             entry[] = None
             self.size -= 1
-            return entry_value.value^
+            _ = entry_value.key^
+            var value = entry_value.value^
+            __mlir_op.`lit.ownership.mark_destroyed`(
```

Comment:
@Mogball @lattner is this the "blessed" way to do this right now?

Reply:
i imagine it's not since the test_reversed still failed... maybe it's not related, but i was really hoping it was

Reply:
it did fix a crash locally for me though. (the exmaple given in #2756)